evm_arithmetization/src/cpu/kernel/asm/account_code.asm

```diff
@@ -139,3 +139,66 @@ remove_padding_loop:
 remove_padding_after:
     %stack (offset, ctx, retdest) -> (retdest, offset)
     JUMP
+
+// Convenience macro to call poseidon_hash_code_unpadded and return where we left off.
+%macro poseidon_hash_code_unpadded
+    %stack (addr, len) -> (addr, len, %%after)
+    %jump(poseidon_hash_code_unpadded)
+%%after:
+%endmacro
+
+/// Applies the padding rule to the code located at the provided address before hashing it.
+/// Memory cells after the last code byte will be overwritten.
+global poseidon_hash_code_unpadded:
+    // stack: addr, len, retdest
+    DUP2 ISZERO %jumpi(poseidon_empty_code)
+    DUP2 DUP2 ADD
+    // stack: padding_addr, addr, len, retdest
+
+    // write 1 after the last code byte
+    DUP1 PUSH 1 MSTORE_GENERAL
+    // stack: padding_addr, addr, len, retdest
+    %increment
+    // stack: padding_addr, addr, len, retdest
+
+    // Pad with 0s until the length is a multiple of 56
+    PUSH 56
+    DUP4 %increment
+    // stack: curr_len, 56, padding_addr, addr, len, retdest
+    MOD
+    // stack: remainder, padding_addr, addr, len, retdest
+    PUSH 56 SUB
```

Comment:
If `curr_len` is 0, I think you would add 56 bytes of padding, which isn't what we want.